soroban-sdk/src/testutils/arbitrary.rs

```diff
@@ -317,19 +317,42 @@ mod objects {
     use super::composite::ArbitraryVal;
     use crate::env::FromVal;
     use crate::ConversionError;
-    use crate::{Env, IntoVal, TryFromVal};
+    use crate::{Env, IntoVal};
 
     use crate::xdr::{Int256Parts, ScVal, UInt256Parts};
     use crate::{
         Address, Bytes, BytesN, Duration, Map, String, Symbol, Timepoint, Val, Vec, I256, U256,
     };
-    use soroban_env_host::TryIntoVal;
+    use soroban_env_host::{TryFromVal, TryIntoVal};
 
     use std::string::String as RustString;
     use std::vec::Vec as RustVec;
 
     //////////////////////////////////
 
+    #[derive(Arbitrary, Debug, Clone, Eq, PartialEq, Ord, PartialOrd)]
+    pub struct ArbitraryOption<T>(Option<T>);
+
+    impl<T> SorobanArbitrary for Option<T>
+    // Problem                      ^ here
```

Comment:
I've got an error on this line here:

```
error[E0277]: the trait bound `soroban_env_host::Val: soroban_env_host::TryFromVal<env::Env, T>` is not satisfied
   --> soroban-sdk/src/testutils/arbitrary.rs:336:34
    |
336 |     impl<T> SorobanArbitrary for Option<T>
    |                                  ^^^^^^^^^ the trait `soroban_env_host::TryFromVal<env::Env, T>` is not implemented for `soroban_env_host::Val`
    |
    = note: required for `T` to implement `soroban_env_host::TryIntoVal<env::Env, soroban_env_host::Val>`
    = note: required for `soroban_env_host::Val` to implement `soroban_env_host::TryFromVal<env::Env, std::option::Option<T>>`
note: required for `soroban_env_host::Val` to implement `env::FromVal<env::Env, std::option::Option<T>>`
   --> soroban-sdk/src/env.rs:98:30
    |
98  | impl<E: internal::Env, T, U> FromVal<E, T> for U
    |                              ^^^^^^^^^^^^^     ^
99  | where
100 |     U: TryFromVal<E, T>,
    |        ---------------- unsatisfied trait bound introduced here
note: required for `std::option::Option<T>` to implement `env::IntoVal<env::Env, soroban_env_host::Val>`
   --> soroban-sdk/src/env.rs:107:30
    |
107 | impl<E: internal::Env, T, U> IntoVal<E, T> for U
    |                              ^^^^^^^^^^^^^     ^
108 | where
109 |     T: FromVal<E, Self>,
    |        ---------------- unsatisfied trait bound introduced here
note: required by a bound in `testutils::arbitrary::api::SorobanArbitrary`
   --> soroban-sdk/src/testutils/arbitrary.rs:222:44
    |
221 |     pub trait SorobanArbitrary:
    |               ---------------- required by a bound in this trait
222 |         TryFromVal<Env, Self::Prototype> + IntoVal<Env, Val> + TryFromVal<Env, Val>
    |                                            ^^^^^^^^^^^^^^^^^ required by this bound in `SorobanArbitrary`
help: consider extending the `where` clause, but there might be an alternative better way to express this requirement
    |
339 |         T: SorobanArbitrary, soroban_env_host::Val: soroban_env_host::TryFromVal<env::Env, T>
    |                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
`Option<T>` implements conversion to and from `Val` in the `soroban-env-common` crate. What am I missing?

Reply:
@brson Do you happen to have any ideas? I'm not sure if this is arising out of me implementing the arbitrary parts of this incorrectly. Any assistance is much appreciated.

Reply:
I think I figured out the `SorobanArbitrary` part in 6666a7b. 🎉

Reply:
Ah I missed this while you were hacking on it. Your fix looks plausible, though the bounds are a surprising mouthful.

Kudos for figuring out those generic type bounds though - I don't know if I could come up with those.

I'm a bit suspicous of the bounds since the bounds for other containers are not so complex. Probably one reason the bounds are complex is that your `try_from_val` impl is propagating errors in conversions from `T::Prototype` to `T`, while the other container types all just assume that conversion will succeed (and I don't think there are any cases where the prototype will not convert).

I will poke at this a bit tomorrow and see if I can arrive at any simplifications.

Reply:
Any simplifications are welcome! To be honest, I got very lost, and just started adding constraints the compiler was telling me to add until it worked. If the PR merges before you take a look, please feel free to open modifications.